module/Company/src/Company/Model/Company.php

```diff
@@ -241,65 +233,168 @@ public function setPhone($phone)
+        return $jobcount;
     }
 
     /**
-     * Set the company's description.
+     * Add a packet to the company.
      *
-     * @param string $description
+     * @param CompanyPacket $packet
      */
-    public function setDescription($description)
+    public function addPacket(CompanyPacket $packet)
     {
-        $this->description = $description;
+        $this->packets->add($packet);
     }
-
+    
     /**
-     * Get the company's jobs.
+     * Remove a packet from the company.
+     * 
+     * @param CompanyPacket $packet packet to remove
+     */
+    public function removePacket(CompanyPacket $packet) 
+    {
+        $this->packets->removeElement($packet);
+    }
+    
+    /**
+     * Get the company's language.
      *
-     * @return Job[]
+     * @return Integer
      */
-    public function getJobs()
+    public function getLanguageNeutralId()
     {
-        return $this->jobs;
+        return $this->languageNeutralId;
     }
-
     /**
-     * Set the company's jobs.
+     * Set the company's language neutral id.
      *
-     * @param Job[] $jobs
+     * @param Integer $languageNeutralId
      */
-    public function setJobs($jobs)
+    public function setLanguageNeutralId($language)
     {
-        $this->jobs = $jobs;
+        $this->languageNeutralId = $language;
+    }
+    // For zend2 forms
+    public function getArrayCopy()
+    {
+        $arraycopy = get_object_vars($this);
+        //$arraycopy["languages"] = new ArrayCollection();
+        $arraycopy["languages"] = Array();
+        foreach($this->getTranslations() as $translation){
+            $arraycopy[$translation->getLanguage()."_"."slogan"] = $translation->getSlogan();
+            $arraycopy[$translation->getLanguage()."_"."website"] = $translation->getWebsite();
+            $arraycopy[$translation->getLanguage()."_"."description"] = $translation->getDescription();
+            $arraycopy[$translation->getLanguage()."_"."logo"] = $translation->getLogo();
+            $arraycopy["languages"][] = $translation->getLanguage();
+        }
+
+        return $arraycopy;
     }
 
+    
+    public function getTranslationFromLocale($locale, $create=false) {
+
+                $companyLanguages =  $this->getTranslations()->map(function ($value){
+                    return $value->getLanguage();
+                } );
+                if(!$companyLanguages->contains($locale)){
+                    if ($create){
+                        $translation = new CompanyI18n();
+                        $translation->setLanguage($locale);
+                        $translation->setCompany($this);
+                    }
+                    else{
+                        $translation = null;
+                    }
+                }
+                else{
+                    $translation = $this->getTranslations()[$companyLanguages->indexOf($locale)];
+                }
+                return $translation;
+    }
+
+    
+    public function exchangeArray($data) {
```

Comment:
This is a long method, that does not have documentation. While it is unclear what it is doing right now.